### PR TITLE
Match files inclusively

### PR DESCRIPTION
**Pull request by rferrazz**
The file is added to the list if it matches any of the provided expressions

### CODE REVIEW

**Comment by thp** [comment]
If no filters were defined, this would return no items, thereby changing default behavior. Granted, there might be a use case for this, but maybe we can expose the desired behavior in some other way? (or at least via a boolean flag that toggles between old and new behavior)

**Comment by rferrazz** [comment]
I've added the filterMode property to switch between exclusive and inclusive filtering

**Comment by thp** [comment]
LGTM. Maybe have a second opinion from @pvuorela and/or @rburchell before merging.

**Comment by pvuorela** [comment]
LGTM

File: src/dirmodel.cpp [hunk 1]
```diff
@@ -282,11 +283,14 @@ void DirModel::onItemsAdded(const QVector<QFileInfo> &newFiles)
...
         bool doAdd = true;
+        if (mFilterMode == Inclusive)
+            doAdd = false;
+
         foreach (const QString &nameFilter, mNameFilters) {
             // TODO: using QRegExp for wildcard matching is slow
             QRegExp re(nameFilter, Qt::CaseInsensitive, QRegExp::Wildcard);
-            if (!re.exactMatch(fi.fileName())) {
-                doAdd = false;
+            if (re.exactMatch(fi.fileName())) {
```
Comment (thp) [inline src/dirmodel.cpp]: Hm, thinking about this, assuming `mFilterMode == Exclusive`, this changes behavior from before:

* Before: `doAdd` is set to `false` iff `!re.exactMatch(fi.fileName())`
* Now: `doAdd` is set to `false` iff `re.exactMatch(fi.fileName())`

I think this should be something like this:

```
bool matched = re.exactMatch(fi.fileName());
if (mFilterMode == Inclusive && matched) {
    doAdd = true;
    break;
} else if (mFilterMode == Exclusive && !matched) {
    doAdd = false;
    break;
}
```

**Comment by thp** [comment]
Merged + tagged as 0.0.12. I've fixed the inclusion/exclusion logic in a separate commit on top of your commits and created a corresponding Mer bug number: https://bugs.merproject.org/show_bug.cgi?id=929

**Pull request merged**